File: NumberAllDivisors/NumberAllDivisors.pas
###

var divisors : integer → List<integer> := number → begin
  number := abs(number);
  var dv := new List<integer>;
  dv.AddRange(|1, number|);
  if (number > 3) then begin
    var k := 2;
    while k * k <= number do begin
      if (number mod k = 0) then
        if number div k > k then dv.AddRange(|k, number div k|) else dv.Add(k);
      inc(k);
    end;
  end;
  dv.Sort;
  result := dv;
end;

// example 
print(divisors(readinteger));

File: NumberAllDivisors/NumberAllDivisors.py
def divisors(number : int) -> []:
    dv = [1, number]
    if number > 3:
        k = 2
        while k * k <= number:
            if number % k == 0:
                if number // k > k:
                    dv.append(k)
                    dv.append(number // k)
                else:
                    dv.append(k)
            k = k + 1
    dv.sort()
    return dv

# example
print(divisors(int(input())))
